src/transformers/generation/flax_utils.py

```diff
@@ -751,6 +751,7 @@ def _beam_search(
         trace: bool = True,
         params: Optional[Dict[str, jnp.ndarray]] = None,
         num_return_sequences: Optional[int] = None,
+        num_beam_groups: Optional[int] = 1,
```

Comment:
In PyTorch we define a separate beam search method for group beam search: 
https://github.com/huggingface/transformers/blob/33b5ef5cdf403ae42f7cdddf906e39073583f062/src/transformers/generation/utils.py#L3375

We only trigger this method if `num_beam_groups>1`:
https://github.com/huggingface/transformers/blob/33b5ef5cdf403ae42f7cdddf906e39073583f062/src/transformers/generation/utils.py#L1426

My opinion is that we should have a separate group beam search method in Flax as well, rather than adding to the existing one. IMO this is cleaner for the reader and more compartmentalised

cc @gante as well for Flax generate design decision

Reply:
> My opinion is that we should have a separate group beam search method in Flax as well, rather than adding to the existing one.

+1 :) 

(btw, there was a [recent bugfix on the PT side](https://github.com/huggingface/transformers/pull/24407), might be relevant here)

Reply:
Awesome, sounds good @yeandy! Excited to see how this pans out!